core/store/src/trie/iterator.rs

```diff
@@ -451,7 +459,7 @@ mod tests {
 
     fn test_seek(trie: &Trie, map: &BTreeMap<Vec<u8>, Vec<u8>>, seek_key: &[u8]) {
         let mut iterator = trie.iter().unwrap();
-        iterator.seek(&seek_key).unwrap();
+        iterator.seek_prefix(&seek_key).unwrap();
         let result1: Vec<_> = iterator.map(Result::unwrap).take(5).collect();
```

Comment:
How about:
```suggestion
        let iterator = iterator.map(Result::unwrap).inspect(|(key, _)| {
            assert!(key.starts_with(seek_key),
                    "‘{key:?x}’ does not start with ‘{start_key:?x}’");
        });
        let result1: Vec<_> = iterator.take(5).collect();
```
And at the end of the function:
```rust
        // Check that all returned keys are prefixed by requested seek key.
        let _ = iterator.count();
```

Reply:
`.take()` consumes `self`.

Reply:
Collect all and truncate vector afterwards then I suppose.

Reply:
where? iterator can't be used after the `take`

Reply:
>  "‘{key:?x}’ does not start with ‘{start_key:?x}’");
this formatting isn't correct.

Reply:
Yeah, I never remember.  It’s either `:?x` or `:x?`.  If the former doesn’t work than it's the latter.